stablehlo/dialect/StablehloOps.cpp

```diff
@@ -183,6 +183,26 @@ LogicalResult TypeExtensionsAttr::verifyEncoding(
       getBounds(), RankedTensorType::get(bounds, elementType), emitError);
 }
 
+llvm::SmallVector<int64_t> getBounds(Type type) {
+  llvm::SmallVector<int64_t> bounds;
+  auto rankedTy = type.dyn_cast<RankedTensorType>();
+  if (rankedTy)
+    if (auto encoding =
+            rankedTy.getEncoding().dyn_cast_or_null<TypeExtensionsAttr>())
+      bounds = llvm::to_vector<4>(encoding.getBounds());
+  return bounds;
+}
+
+void addBounds(MLIRContext* context, SmallVectorImpl<Type>& types,
+               llvm::SmallVector<int64_t> bounds) {
+  if (!bounds.empty()) {
+    auto type = types.pop_back_val().cast<RankedTensorType>();
+    Attribute encoding = TypeExtensionsAttr::get(context, bounds);
+    types.emplace_back(RankedTensorType::get(type.getShape(),
+                                             type.getElementType(), encoding));
+  }
+}
+
```

Comment:
This is why we have `BoundedAttrInterface` and `BoundedDialectInterface::createBoundedAttr` (see example of their usage [here](https://github.com/openxla/stablehlo/blob/424819cc8939b9db1f7d02dbd76f8358a9e234ef/stablehlo/dialect/Base.h#L218-L289)). I think they should be sufficient to avoid passing the bounds around.

Reply:
Thanks for this example! Change. 
Revert a lot of change so that comparing to HEAD is more easier to review.